django/src/rdwatch/models/site_evaluation.py

```diff
@@ -19,6 +19,10 @@ class SiteEvaluation(models.Model):
     id = models.UUIDField(primary_key=True, default=uuid4, editable=False)
 
     timestamp = CreationDateTimeField()
+    modified_timestamp = models.DateTimeField(
+        auto_now=True,
+        help_text='Timestamp of the last modification',
+    )
```

Comment:
Using a modified timestamp makes sense logically, but unfortunately there is a major issue with the `auto_now` option. Django sort of implies that this is a magic column that will always get updated when the row is modified, but sadly that's not the case - `modified_timestamp` is updated when calling .save() on a model, but it will not be updated if .update() is used. See https://docs.djangoproject.com/en/5.0/ref/models/fields/#django.db.models.DateField.auto_now

When we tried using this feature on other projects this ended up causing some weird data inconsistency bugs as the codebase got more and more complex. In the end we ended up having rewriting things to remove it and explicitly update the `modified_timestamp` field when needed to remove any ambiguity. I don't think we're necessarily at that point yet (and maybe we'll never get there, since at the end of the day these are only used for caching purposes, it's not like they're user facing or there's any other functionality depending on them), but I think we should consider using the explicit update approach instead - any thoughts?

Reply:
I am all for being explicit if there may be some confusion.  I've left it as  standard DateTime field, added a comment about manually modifying it.  Then added the updates to the creationTime and the patch endpoints for siteEvaluations.